src/tmpfiles/tmpfiles.c

```diff
@@ -903,16 +918,18 @@ static int dir_cleanup(
         if (deleted && (self_atime_nsec < NSEC_INFINITY || self_mtime_nsec < NSEC_INFINITY)) {
                 struct timespec ts[2];
 
-                log_debug("Restoring access and modification time on \"%s\": %s, %s",
-                          p,
-                          FORMAT_TIMESTAMP_STYLE(self_atime_nsec / NSEC_PER_USEC, TIMESTAMP_US),
-                          FORMAT_TIMESTAMP_STYLE(self_mtime_nsec / NSEC_PER_USEC, TIMESTAMP_US));
+                log_action("Would restore", "Restoring",
+                           "%s access and modification time on \"%s\": %s, %s",
+                           p,
+                           FORMAT_TIMESTAMP_STYLE(self_atime_nsec / NSEC_PER_USEC, TIMESTAMP_US),
+                           FORMAT_TIMESTAMP_STYLE(self_mtime_nsec / NSEC_PER_USEC, TIMESTAMP_US));
 
                 timespec_store_nsec(ts + 0, self_atime_nsec);
                 timespec_store_nsec(ts + 1, self_mtime_nsec);
 
                 /* Restore original directory timestamps */
-                if (futimens(dirfd(d), ts) < 0)
+                if (!arg_dry_run &&
+                    futimens(dirfd(d), ts) < 0)
```

Comment:
this should not be conditioned out I guess. people assume --dry-run has no effect on system behaviour, but it *has* because you touch all atimes here. Hence I think if you enter the dirs, you better restore the atimes afterwards, even if --dry-run is used.

also the log message above then needs updates. (should probably stay unconditionally at debug)

Reply:
hmm, wasn't the problem with this that O_NOATIME only works for root? hence a manual reset is still necessary?

(i mean, we could suppress it if it turns out to be unnecessary if the the timestamp already matches the status quo ante)

Reply:
ah, so O_NOATIME is restricted to super user or owner of the file. And I guess if you are neither then you couldn't call futimens() either on the thing, hence we probably should be good without it i figure